### PR TITLE
Docker Support

### LINKED ISSUE #10: Docker Compose
Add `Dockerfile` and a Docker Compose setup to automatically setup `redis` and `mysql-server`, maybe other services like `elasticsearch` later.

### PR DESCRIPTION
**Pull request by kelunik**
Closes #10. Still needs some work, e.g. `sleep(5);` to wait for MySQL.

### CODE REVIEW

File: res/config/config.sample.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    /* https://developers.google.com/recaptcha/docs/faq */
+    "recaptchaKey" => "6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI",
+    "recaptchaSecret" => "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe",
+];
+
+// Wait for MySQL to start up
+sleep(5);
+
+echo `mysql --host mysql -uroot -proot < /app/res/database/schema.sql`;
+echo `mysql --host mysql -uroot -proot bugcache < /app/res/database/init.sql`;
```
Comment (bwoebi) [inline res/config/config.sample.php]: I don't feel like you should put this into the config.sample.php file ... put it in a separate file to be used with docker only

**Comment by bwoebi** [comment]
Can we please keep config files in the root? That's where I'd intuitively search for them...

**Comment by kelunik** [comment]
No. Our root is growing and growing. I like to keep files out of it that don't need to be in the root.

**Comment by bwoebi** [comment]
yeah, I'd be fine with moving the Dockerfile somewhere else, but the config _really_ should remain inside the root.

**Comment by kelunik** [comment]
The `Dockerfile` should absolutely stay there, as it's the standard path for it and otherwise requires addional arguments.

But the config doesn't usually matter in dev, since it's anyway executed via Docker.

All we need is a live reload of Aerys, do we have that in debug mode yet?

**Comment by bwoebi** [comment]
That is, if you execute in Docker. You may not want to. (And certainly not when deploying.)

Keep it in the root, _please_.

Also, no, we don't have live reload in debug mode. Kill and restart is the way to go there.

**Comment by kelunik** [comment]
Running in Docker will be the recommend and preferred way. Also it's perfectly fine for staging and small production sites.

**Comment by bwoebi** [comment]
Staging should be as close to production as possible, so no.

Running in docker is fine if you develop on it, but for everything else, rather not.

It is a nice to have, but nothing more.

**Comment by kelunik** [comment]
> Running in docker is fine if you develop on it, but for everything else, rather not.

Why rather not?

**Comment by bwoebi** [comment]
It doesn't integrate well, if you run multiple apps with one aerys instance (yes, yes …).
Also some overhead (don't know how large) and less flexible (you may want to have non-default credentials for mysql (e.g. to hide secbugs to other users))

I don't see many upsides apart from one a bit easier initial configuration...

**Comment by kelunik** [comment]
It doesn't integrate well, if you run multiple apps with one aerys instance (yes, yes …).

> Yes, but that doesn't matter. In a typical production environment you currently need a proxy in front of it anyway, e.g. for OCSP stapling.

> Also some overhead (don't know how large)

Negligible to non-existent.
 
> and less flexible (you may want to have non-default credentials for mysql (e.g. to hide secbugs to other users))

Perfectly fine with your own `docker-compose.yml`

> I don't see many upsides apart from one a bit easier initial configuration...

Completely easy configuration instead of setting up a bunch of tools. Also when things are updated. Also the separation of dependencies like the MySQL server. It's just the same reason why you don't install all Composer dependencies globally.

**Pull request merged**